docs/src/counting.md

```diff
@@ -6,66 +6,73 @@ end
 ```
 
 # Counting
+`BioSequences` contain functionality to efficiently count biosymbols in a biosequence
+that satisfies some predicate.
+
+Consider a naive counting function like this:
+```julia
+function count_Ns(seq::BioSequence{<:DNAAlphabet})
+    ns = 0
+    for i in seq
+        ns += (i == DNA_N)::Bool
+    end
+    ns
+end 
+```
 
-BioSequences extends the `Base.count` method to provide some useful utilities for
-counting the number of sites in biological sequences.
-
-Most generically you can count the number of sites that satisfy some condition
-i.e. cause some function to return `true`:
+This function can be more efficient implemented by exploiting the internal
+data encoding of certain biosequences.
+Therefore, Julia provides optimised methods for `Base.count`, such that `count_Ns`
+above can be more efficiently expressed `count(==(DNA_N), seq)`.
 
-```jldoctest
-julia> count(isambiguous, dna"ATCGM")
-1
+!!! note
+    It is important to understand that this speed is achieved with custom methods of
+    `Base.count`, and not by a generic mechanism that improves the speed of counting
+    symbols in `BioSequence `in general.
+    Hence, while `count(==(DNA_N), seq)` may be optimised,
+    `count(i -> i == DNA_N, seq)` is not, as this is a different method.
 
-```
+## Currently optimised methods
+By default, only `LongSequence` and `LongSubSeq` have optimised methods. Downstream
+implementers of new `BioSequence`s may not be optimised.
```

Comment:
This sort of sounds like you're saying they *can't* be optimized, but I think you're saying it's possible that they will, but also possible that they won't. 

"May not" in English is ambiguous - it sometimes means "might not" and sometimes means "not allowed to be". 

```suggestion
implementers of new `BioSequence`s might not be optimised.
```

Reply:
Actually, "might not" actually _can_ also mean "not able to", but it is rarely used that way in practice. English is really weird.